### PR TITLE
#12661 - Checkbox trees don't trigger unsaved changes warning and can…

### LINKED ISSUE #12661: Checkbox trees don't trigger unsaved changes warning and can't be discarded
### Bug Description

Changes made to the checkboxes for 'Weather conditions at time of sampling' are not considered for the triggering of the 'Confirm navigation' pop-up and are not discarded when the user clicks 'Discard' at the bottom of the page.

### Reproduction Steps

I - 'Confirm navigation' pop-up not triggered:
1. Log in with an Environmental Surveillance User;
2. Open an existing environment sample;
3. Check some of the checkboxes from 'Weather conditions at time of sampling';
4. Click to navigate to any other page in the app.
**Actual behavior:** The confirm navigation pop-up is not triggered by unsaved changes made to the checkboxes.

II - Changes made are not discarded:
1. Log in with an Environmental Surveillance User;
2. Open an existing environment sample;
3. Check some of the checkboxes from 'Weather conditions at time of sampling';
4. Click 'Discard' at the bottom of the page.
**Actual behavior:** The changes made to the checkboxes are not discarded.

### Expected Behavior

Expected behavior for scenario I: Any unsaved change on the environment sample page, including to the checkboxes, should trigger the 'Confirm navigation' pop-up when the user tries to navigate to another page of the app.

Expected behavior for scenario II: Any unsaved changes at the time when the user clicks 'Discard' should be discarded.

### Affected Area(s)

Web app

### SORMAS Version

1.90.0, 1.89.0

### Android version/Browser

Chrome

### User Role(s)

Environmental Surveillance User

### Server URL

local machine 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
…'t be discarded - fix checks

Fixes #12661

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13020

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTree.java [hunk 1]
```diff
@@ -208,11 +212,9 @@ public void setValue(Map<ENUM, Boolean> newFieldValue) throws ReadOnlyException,
 	}
 
 	private void setToggleValues(Map<ENUM, Boolean> newFieldValue) {
-		settingToggles = true;
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTree.java:211]: removing the flag changes here makes `setValue` to be called recoursively for each checkbox, that again triggers setting the checkboxes and could result in inifite loop

the solution for the issue it to don't call setValue in the chaeckobx change listener while settingToggles is true.

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13020

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTree.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTree.java]: changed

**Pull request merged**